src/main/java/com/softserveinc/ita/pageobjects/BasePage.java

```diff
@@ -80,15 +80,29 @@ public boolean isLanguageSwitchedTo(LanguageSwitcher language) {
         return searchButtonText.equals(verificationWord);
     }
 
-    public T selectRequiredCategory(String categoryName) {
+    public BasePage<T> selectRequiredCategory(String categoryName) {
         $x("//a[@class ='menu-categories__link' and contains(text(),'" + categoryName + "')]").click();
 
-        return (T) this;
+        return this;
     }
 
     public ProductPage selectRandomSubCategory() {
-        randomizerForListCategories($$x("//*[@class='tile-cats__heading tile-cats__heading_type_center ng-star-inserted']"));
+        var list = $$x("//*[@Class='tile-cats__heading tile-cats__heading_type_center ng-star-inserted']")
+                .shouldBe(sizeNotEqual(0), ofSeconds(10));
+        list
+                .get(getRandomNumber(list.size()))
+                .click();
 
         return new ProductPage();
     }
+
+    public BasePage<T> closeAdBanner() {
```

Comment:
T instead BasePage<T>

is not a duplicate of this method  closeAdvertisingBannerIfDisplayed?

Reply:
i didnt saw it. Probably Pavlo did it too